examples/step-33/step-33.cc

```diff
@@ -1607,7 +1607,11 @@ namespace Step33
       }
 
     // After all this assembling, notify the Trilinos matrix and vector object
-    // that the linear system is done:
+    // that the linear system is done by invoke the function compress(). However,
+    // these two function calls here have no actual influence and you can remove
+    // them safely. The reason is this tutorial program is not designed to run in
+    // distributed parallel mode. See @ref GlossCompress "Compressing distributed objects"
+    // for more information on what is compress() designed to do.
```

Comment:
> 在 2015年4月12日，下午4:03，Wolfgang Bangerth <notifications@github.com> 写道：
> 
> In examples/step-33/step-33.cc <https://github.com/dealii/dealii/pull/786#discussion_r28207411>:
> 
> > @@ -1607,7 +1607,11 @@ namespace Step33
> >        }
> >  
> >      // After all this assembling, notify the Trilinos matrix and vector object
> > -    // that the linear system is done:
> > +    // that the linear system is done by invoke the function compress(). However,
> > +    // these two function calls here have no actual influence and you can remove
> > +    // them safely. The reason is this tutorial program is not designed to run in
> > +    // distributed parallel mode. See @ref GlossCompress "Compressing distributed objects"
> > +    // for more information on what is compress() designed to do.
> I think if you have to go to lengths explaining why something is unnecessary, then maybe removing the whole piece is the right thing to do :-)
> 
> 

But I think the people those who interesting in Euler equations are highly possible to parallelize these code, just like me. Then this piece of code will save them a head scratching problem.
So, I think keep the two compress() would be better than remove them.

Then, If we leave the two compress() without explanation, people may confuse them with compress a full matrix into a sparse one — which is also my first reaction on seeing system_matrix.compress(). Then its natural to think, why a vector also need to compress?

After all, those who do not care about parallelization could find out they are safe to ignore this very quickly.